human_cancer_analysis/nbconverted/2_identify_generic_genes_pathways.py

```diff
@@ -0,0 +1,778 @@
+
+
+# In[18]:
+
+
+# Take absolute value of logFC and t statistic
+simulated_DE_stats_all = process.abs_value_stats(simulated_DE_stats_all)
+
+
+# In[19]:
+
+
+# Aggregate statistics across all simulated experiments
+simulated_DE_summary_stats = calc.aggregate_stats(
+    col_to_rank_genes,
+    simulated_DE_stats_all,
+    'DE'
+)
+
+
+# In[20]:
+
+
+# Take absolute value of logFC and t statistic
+template_DE_stats = process.abs_value_stats(template_DE_stats)
+
+# Rank genes in template experiment
+template_DE_stats = calc.rank_genes_or_pathways(
+    col_to_rank_genes,      
+    template_DE_stats,
+    True
+)
+
+
+# In[21]:
+
+
+# Rank genes in simulated experiments
+simulated_DE_summary_stats = calc.rank_genes_or_pathways(
+    col_to_rank_genes,
+    simulated_DE_summary_stats,
+    False
+)
+
+
+# ### Gene summary table
+
+# In[22]:
+
+
+summary_gene_ranks = process.generate_summary_table(
+    template_DE_stats,
+    simulated_DE_summary_stats,
+    col_to_rank_genes,
+    local_dir
+)
+
+summary_gene_ranks.head()
+
+
+# In[23]:
+
+
+# Create `gene_summary_fielname`
+summary_gene_ranks.to_csv(gene_summary_filename, sep='\t')
+
+
+# ### Compare gene ranking
+# Studies have found that some genes are more likely to be differentially expressed even across a wide range of experimental designs. These *generic genes* are not necessarily specific to the biological process being studied but instead represent a more systematic change. 
+# 
+# We want to compare the ability to detect these generic genes using our method vs those found by [Crow et. al. publication](https://www.pnas.org/content/pnas/116/13/6491.full.pdf). Their genes are ranked 0 = not commonly DE; 1 = commonly DE. Genes by the number differentially expressed gene sets they appear in and then ranking genes by this score.
+
+# In[24]:
+
+
+# Get generic genes identified by Crow et. al.
+DE_prior_file = params['reference_gene_filename']
+ref_gene_col = params['reference_gene_name_col']
+ref_rank_col = params['reference_rank_col']
+
+figure_filename = f"gene_ranking_{col_to_rank_genes}.svg"
+
+process.compare_gene_ranking(
+    summary_gene_ranks,
+    DE_prior_file,
+    ref_gene_col,
+    ref_rank_col,
+    figure_filename
+)
+
+
+# **Takeaway:**
+# * Previously we compared gene ranks obtained from (recount2)-trained VAE model vs gene ranks obtained from manual curation using Crow et. al data. This [PR](https://github.com/ajlee21/generic-expression-patterns/blob/807377d76f63b6282c62255d7b160feb8585e0e2/human_analysis/2_identify_generic_genes_pathways.ipynb) shows that the correlation of gene ranks are very consistent.
+# 
+# * Here we are comparing gene ranks obtained from a (Powers et. al.)-trained VAE model vs gene ranks obtained from manual curation using Crow et. al. Based on this correlation plot there is a high correlation between those very high and low ranked genes -- high correlation at the extremes but there is a lot of noise in the middle.
```

Comment:
So this is similar to what you found using the recount2 model, right? At least the pair plot looks similar to me (high correlation for very high and very low ranked genes, but noisy in the middle)

Reply:
Yes similar, but there was a bit less noise in the middle and more of a clearer correlation pattern. This correlation is comparing generic genes found using a cancer-specific training set vs generic genes found using a mixed/general training set. So it would seem that the highly and lowly generic genes are well correlated but not much in the middle when we have cross training sets